### PR TITLE
HDXDSYS-985 HNO is now one dataset with potentially multiple resources

### PR DESCRIPTION
**Pull request by mcarans**
You can merge this if needed since without this fix the HNO pipeline will fail.

### CODE REVIEW

File: src/hapi/pipelines/database/humanitarian_needs.py [hunk 1]
```diff
@@ -58,26 +59,24 @@ def get_admin2_ref(self, row, dataset_name, errors):
...
-        )
         warnings = set()
         errors = set()
-        for dataset in datasets:
+        reader = Read.get_reader("hdx")
+        dataset = reader.read_dataset("global-hpc-hno", self._configuration)
+        self._metadata.add_dataset(dataset)
+        dataset_id = dataset["id"]
+        dataset_name = dataset["name"]
+        for resource in dataset.get_resources():
```
Comment (turnerm) [inline src/hapi/pipelines/database/humanitarian_needs.py:71]: Maybe we only need to include the latest resource in HAPI? As we're doing for 3W

File: src/hapi/pipelines/database/admins.py [hunk 2]
```diff
@@ -39,7 +39,7 @@ def __init__(
         self.admin1_data = {}
         self.admin2_data = {}
 
-    def populate(self):
+    def populate(self) -> None:
```
Comment (turnerm) [inline src/hapi/pipelines/database/admins.py:42]: How did you catch all these, are you using a typing extension? Should we add something to pre-commit maybe?

**Review by turnerm — APPROVED** [review]

File: src/hapi/pipelines/database/admins.py (hunk 2 above)
Comment (mcarans) [reply src/hapi/pipelines/database/admins.py:42]: I stumbled across one and then noticed it was like that for all of them. It wasn't anything automated. Such a check could help. mypy is one tool used for type checking that probably could be set up in pre-commit.

File: src/hapi/pipelines/database/humanitarian_needs.py (hunk 1 above)
Comment (mcarans) [reply src/hapi/pipelines/database/humanitarian_needs.py:71]: Yes following from our discussion I realise this

File: src/hapi/pipelines/database/admins.py (hunk 2 above)
Comment (mcarans) [reply src/hapi/pipelines/database/admins.py:42]: This change is something for a separate PR.

**Pull request merged**